src/render.rs

```diff
@@ -0,0 +1,54 @@
+use crate::package::Package;
+use jrsonnet_evaluator::{
+	trace::{ExplainingFormat, PathResolver},
+	EvaluationState, FileImportResolver,
+};
+use serde_json::Value;
+
+const VALUES_PARAM: &str = "values";
+
+pub struct Renderer {
+	state: EvaluationState,
+}
+
+impl Renderer {
+	pub fn new() -> Self {
+		let state = EvaluationState::default();
+		let resolver = PathResolver::Absolute;
+		state.set_trace_format(Box::new(ExplainingFormat { resolver }));
+
+		state.with_stdlib();
+
+		Renderer { state }
+	}
+
+	pub fn render(&self, pkg: &Package) -> Result<Value, String> {
+		let state = &self.state;
+
+		state.set_import_resolver(Box::new(FileImportResolver {
+			library_paths: vec![pkg.root.clone()],
+		}));
+
+		state
+			.add_tla_code(VALUES_PARAM.into(), pkg.values.to_string().into())
+			.map_err(|_err| String::from("Failed to register values parameter"))?;
+
+		let parsed = state
+			.evaluate_file_raw(&pkg.spec.main)
+			.map_err(|err| format!("Failed when evaluating the script: {}", err.error()))?;
+
+		let wrapped = state
+			.with_tla(parsed)
+			.map_err(|err| format!("Failed when evaluating the script: {}", err.error()))?;
+
+		let rendered = state
+			.manifest(wrapped)
+			.map_err(|err| format!("Failed to render the template: {}", err.error()))?
+			.to_string();
+
+		let json = serde_json::from_str(&rendered)
+			.map_err(|err| format!("Failed to parse your results to json: {}", err))?;
```

Comment:
Would be great encapsulate each action in a function!
Something like this:
```rust
fn parse(...) -> ... {
    ...
}

fn wrapped(...) -> ... {
    ...
}
```

Reply:
It doesn't worth it. There's no reason to do it at the current moment.

The function is responsible for rendering a jsonnet function with a tla parameter called `values`, and these are the steps for that. It doesn't have different abstraction levels once I use the `EvaluationState` public API straightforwardly. If we were to split it further, it would create pointless delegates with a signature mirroring the underneath `EvaluationState` but with arbitrary names, such as `wrapped(state, parsed)`.